SDL/Event.cu

```diff
@@ -574,464 +566,151 @@ void SDL::Event::resetObjectsInModule()
-    cudaMemcpyAsync(hitsInGPU->etas,host_etas,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU->idxs,host_idxs,loopsize*sizeof(unsigned int),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU->moduleIndices,host_moduleIndex,loopsize*sizeof(uint16_t),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU->highEdgeXs,host_highEdgeXs,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU->highEdgeYs,host_highEdgeYs,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU->lowEdgeXs,host_lowEdgeXs,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU->lowEdgeYs,host_lowEdgeYs,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU->nHits,&loopsize,sizeof(unsigned int),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-    cudaMemcpyAsync(hitsInGPU->hitRanges,module_hitRanges,nModules*2*sizeof(int),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-    cudaMemcpyAsync(hitsInGPU->hitRangesLower,module_hitRangesLower,nModules*sizeof(int),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-    cudaMemcpyAsync(hitsInGPU->hitRangesUpper,module_hitRangesUpper,nModules*sizeof(int),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-    cudaMemcpyAsync(hitsInGPU->hitRangesnLower,module_hitRangesnLower,nModules*sizeof(int8_t),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-    cudaMemcpyAsync(hitsInGPU->hitRangesnUpper,module_hitRangesnUpper,nModules*sizeof(int8_t),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-cudaStreamSynchronize(stream);
-
-    cms::cuda::free_host(host_rts);
-    cms::cuda::free_host(host_phis);
-    cms::cuda::free_host(host_etas);
-    cms::cuda::free_host(host_moduleIndex);
-    cms::cuda::free_host(host_highEdgeXs);
-    cms::cuda::free_host(host_highEdgeYs);
-    cms::cuda::free_host(host_lowEdgeXs);
-    cms::cuda::free_host(host_lowEdgeYs);
-    cms::cuda::free_host(module_layers);
-    cms::cuda::free_host(module_subdet);
-    cms::cuda::free_host(module_partnerModuleIndices);
-    cms::cuda::free_host(module_hitRanges);
-    cms::cuda::free_host(module_hitRangesLower);
-    cms::cuda::free_host(module_hitRangesUpper);
-    cms::cuda::free_host(module_hitRangesnLower);
-    cms::cuda::free_host(module_hitRangesnUpper);
-    cms::cuda::free_host(module_moduleType);
-    cms::cuda::free_host(host_x);
-    cms::cuda::free_host(host_y);
-    cms::cuda::free_host(host_z);
-    cms::cuda::free_host(host_detId);
-    cms::cuda::free_host(host_idxs);
-
+    // No stream synchronize needed after second kernel call. Saves ~100 us.
+    moduleRangesKernel<<<MAX_BLOCKS,256,0,stream>>>(nLowerModules, modulesInGPU, hitsInGPU);
+    //std::cout << cudaGetLastError() << std::endl;
```

Comment:
Is the sync not needed because the next operation is `addPixelSegmentToEvent`), which does not rely on `hitsInGPU.hitRanges*`  (the only values modified in this kernel)?

It would be better to be clear about this, to reduce a chance of errors in the future